ggshield/core/config/user_config.py

```diff
@@ -320,30 +320,33 @@ def load(cls, config_path: Optional[Path] = None) -> Tuple["UserConfig", Path]:
         Returns a UserConfig instance, and the path where updates should be saved
         """
         deprecation_messages: List[str] = []
-        if config_path:
-            logger.debug("Loading custom config from %s", config_path)
-            dct = _load_config_dict(config_path, deprecation_messages)
-            user_config = UserConfig.from_config_dict(dct)
-            user_config.deprecation_messages = deprecation_messages
-            return user_config, config_path
-
-        user_config_dict: Dict[str, Any] = {}
-        global_config_path = find_global_config_path()
-        if global_config_path:
-            dct = _load_config_dict(global_config_path, deprecation_messages)
-            update_dict_from_other(user_config_dict, dct)
-            logger.debug("Loaded global config from %s", global_config_path)
-        else:
-            logger.debug("No global config")
-
-        local_config_path = find_local_config_path()
-        if local_config_path:
-            dct = _load_config_dict(local_config_path, deprecation_messages)
-            update_dict_from_other(user_config_dict, dct)
-            config_path = local_config_path
-            logger.debug("Loaded local config from %s", local_config_path)
-        else:
-            logger.debug("No local config")
+        try:
+            if config_path:
+                logger.debug("Loading custom config from %s", config_path)
+                dct = _load_config_dict(config_path, deprecation_messages)
+                user_config = UserConfig.from_config_dict(dct)
+                user_config.deprecation_messages = deprecation_messages
+                return user_config, config_path
+
+            user_config_dict: Dict[str, Any] = {}
+            global_config_path = find_global_config_path()
+            if global_config_path:
+                dct = _load_config_dict(global_config_path, deprecation_messages)
+                update_dict_from_other(user_config_dict, dct)
+                logger.debug("Loaded global config from %s", global_config_path)
+            else:
+                logger.debug("No global config")
+
+            local_config_path = find_local_config_path()
+            if local_config_path:
+                dct = _load_config_dict(local_config_path, deprecation_messages)
+                update_dict_from_other(user_config_dict, dct)
+                config_path = local_config_path
+                logger.debug("Loaded local config from %s", local_config_path)
+            else:
+                logger.debug("No local config")
+        except AttributeError:
+            raise UnexpectedError("Failed to load user configuration.")
```

Comment:
If it's possible, I would prefer if this was handled in `update_dict_from_other()`:
- we can include in the error message the name of the failing key
- it avoids having a long block of code inside the try...except block

Reply:
I understand what you mean, let me know if this new version works for you. 

I initally made this choice because `update_dict_from_other()` is a utils function, so I wanted to avoid raising specific errors for a function that could be reused.

Reply:
That was good reasoning: I was worried about `update_dict_from_other()` being an utils function as well, but then I saw it was only used for config, and that it was in core.config.utils, so we can accept this specialization.